File: minimax/ios/DreamEcho/DreamEcho/Models/Podcast.swift
import Foundation

// MARK: - Podcast Model

struct Podcast: Codable, Identifiable, Hashable {
    let id: String
    let title: String
    let description: String
    let imageURL: String?
    let author: String
    let rssURL: String
    let link: String?
    
    enum CodingKeys: String, CodingKey {
        case id, title, description
        case imageURL = "image_url"
        case author
        case rssURL = "rss_url"
        case link
    }
    
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        self.id = UUID().uuidString
        self.title = try container.decode(String.self, forKey: .title)
        self.description = try container.decode(String.self, forKey: .description)
        self.imageURL = try container.decodeIfPresent(String.self, forKey: .imageURL)
        self.author = try container.decodeIfPresent(String.self, forKey: .author) ?? ""
        self.rssURL = try container.decode(String.self, forKey: .rssURL)
        self.link = try container.decodeIfPresent(String.self, forKey: .link)
    }
    
    init(title: String, description: String, imageURL: String?, author: String, rssURL: String, link: String?) {
        self.id = UUID().uuidString
        self.title = title
        self.description = description
        self.imageURL = imageURL
        self.author = author
        self.rssURL = rssURL
        self.link = link
    }
    
    func hash(into hasher: inout Hasher) {
        hasher.combine(rssURL)
    }
    
    static func == (lhs: Podcast, rhs: Podcast) -> Bool {
        lhs.rssURL == rhs.rssURL
    }
}

// MARK: - Episode Model

struct Episode: Codable, Identifiable, Hashable {
    let id: String
    let title: String
    let description: String
    let audioURL: String?
    let pubDate: String
    let duration: String?
    let image: String?
    
    enum CodingKeys: String, CodingKey {
        case id, title, description
        case audioURL = "audio_url"
        case pubDate = "pub_date"
        case duration
        case image
    }
    
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        self.id = UUID().uuidString
        self.title = try container.decode(String.self, forKey: .title)
        self.description = try container.decode(String.self, forKey: .description)
        self.audioURL = try container.decodeIfPresent(String.self, forKey: .audioURL)
        self.pubDate = try container.decode(String.self, forKey: .pubDate)
        self.duration = try container.decodeIfPresent(String.self, forKey: .duration)
        self.image = try container.decodeIfPresent(String.self, forKey: .image)
    }
    
    init(title: String, description: String, audioURL: String?, pubDate: String, duration: String?, image: String?) {
        self.id = UUID().uuidString
        self.title = title
        self.description = description
        self.audioURL = audioURL
        self.pubDate = pubDate
        self.duration = duration
        self.image = image
    }
    
    func hash(into hasher: inout Hasher) {
        hasher.combine(title)
    }
    
    static func == (lhs: Episode, rhs: Episode) -> Bool {
        lhs.title == rhs.title && lhs.audioURL == rhs.audioURL
    }
}

// MARK: - Feed Response

struct FeedResponse: Codable {
    let title: String
    let description: String
    let image: String?
    let author: String?
    let link: String?
    let items: [Episode]
    
    enum CodingKeys: String, CodingKey {
        case title, description, image, author, link, items
    }
}

// MARK: - User Model

struct User: Codable, Identifiable {
    let id: String
    let username: String
}

// MARK: - Auth Response

struct AuthResponse: Codable {
    let token: String
    let user: User
}

File: android/app/src/main/kotlin/com/dreamecho/app/models/Podcast.kt
package com.dreamecho.app.models

import com.google.gson.annotations.SerializedName

// Podcast data model
data class Podcast(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("title")
    val title: String,
    
    @SerializedName("description")
    val description: String,
    
    @SerializedName("image_url")
    val imageUrl: String?,
    
    @SerializedName("author")
    val author: String,
    
    @SerializedName("rss_url")
    val rssUrl: String,
    
    @SerializedName("link")
    val link: String?
)

// Episode data model
data class Episode(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("title")
    val title: String,
    
    @SerializedName("description")
    val description: String,
    
    @SerializedName("audio_url")
    val audioUrl: String?,
    
    @SerializedName("pub_date")
    val pubDate: String,
    
    @SerializedName("duration")
    val duration: String?,
    
    @SerializedName("image")
    val image: String?
)

// Feed response from API
data class FeedResponse(
    @SerializedName("title")
    val title: String,
    
    @SerializedName("description")
    val description: String,
    
    @SerializedName("image")
    val image: String?,
    
    @SerializedName("author")
    val author: String?,
    
    @SerializedName("link")
    val link: String?,
    
    @SerializedName("items")
    val items: List<Episode>
)

// User model
data class User(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("username")
    val username: String
)

// Auth response
data class AuthResponse(
    @SerializedName("token")
    val token: String,
    
    @SerializedName("user")
    val user: User
)

// Login request
data class LoginRequest(
    @SerializedName("username")
    val username: String,
    
    @SerializedName("password")
    val password: String
)

